build-packages.sh

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env bash
+# Build logic for KSQLDB & its OS packages
+
+set -o pipefail -o nounset -o errexit
+
+function error {
+    echo "ERROR: ${*}"
+    exit 1
+}
+
+WORKSPACE=""
+FULL_VERSION=""
+UPSTREAM_VERSION=""
+DOCKER_REGISTRY=""
+BUILD_JAR="false"
+while [[ "${#}" -gt 0 ]]; do
+    arg="${1}"
+    case "${arg}" in
+	    -w|--workspace)
+            WORKSPACE="${2}"
+            [ -d "${WORKSPACE}" ] || error "'${WORKSPACE}' Doesn't exist or is not a directory"
+            shift 2
+            ;;
+        -p|--project-version)
+            FULL_VERSION="${2}"  
+            shift 2
+            ;;
+        -u|--upstream-version)
+            UPSTREAM_VERSION="${2}"
+            shift 2
+            ;;
+        -j|--jar)
+            BUILD_JAR="true"
+            shift
+            ;;
+        -d|--docker-registry)
+            DOCKER_REGISTRY="${2}"
+            shift 2
+            ;;
+        *)
+            error "Unknown arg ${arg}"
+            ;;
+    esac
+done
+
+[[ -z "${WORKSPACE}" ]] && error "-w/--workspace not specified and is required"
+[[ -z "${FULL_VERSION}" ]] && error "-p/--project-version not specified and is required"
+[[ -z "${UPSTREAM_VERSION}" ]] && error "-u/--upstream-version not specified and is required"
+[[ -z "${DOCKER_REGISTRY}" ]] && error "-d/--docker-registry not specified and is required"
+ 
+
+if [[ "${FULL_VERSION}" =~ ^[0-9]+.[0-9]+.[0-9]+-rc[0-9]+$ ]]; then
+    RELEASE="0.${FULL_VERSION##*-rc}"
+    VERSION="${FULL_VERSION%%-rc*}"
```

Comment:
nit: can you add some comments describing what each of these are so we don't need to read the code to understand how they're used?

Reply:
I will add something.